File: Python3/409. Longest Palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        chars = Counter(s)
        length = 0
        
        for count in chars.values():
            if count%2 == 0:
                length += count
            elif count > 1:
                length += count-1
                
        return length+1 if length<=len(s)-1 else length
        

File: Java/409. Longest Palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        Map<Character, Integer> chars = new HashMap<>();
        for(char c: s.toCharArray()) {
            chars.put(c, chars.getOrDefault(c, 0)+1);
        }
        
        int count = 0;
        for(int c: chars.values()) {
            if(c%2 == 0) {
                count += c;
            } else if(c > 1) {
                count += c-1;
            }
        }
        
        return count < s.length() ? count+1 : count;
    }
}